baybe/serialization/core.py

```diff
@@ -54,9 +54,43 @@ def get_base_structure_hook(
     # TODO: use include_subclasses (https://github.com/python-attrs/cattrs/issues/434)
     from baybe.utils.basic import get_subclasses
 
-    def structure_base(val: dict, _: type[_T]) -> _T:
+    def structure_base(val: Union[dict, str], _: type[_T]) -> _T:
```

Comment:
I'm not sure if I'm super convinced by that approach, which makes the code quite spaghetti-like and now gives the method an additional responsibility (before, it was only unstructuring the dict, now it is additional conversion logic and it's like 5x as long). Without having thought too much about it, my first impulse would have been to just create a separate hook that replaces the abbreviation with the full name and then just calls the original hook. That way, the logic is cleanly separated and one would apply the hook only where needed. What do you think about it?

Reply:
these changes also include the possibility that you can use now the abbreviation in the `type` field in case of a dictionary too. Tha would be missing by the pure wrapper approach

But it could simply eb added by keeping the lower part of the unction chagnes here and moving the uppper part out to another wrapper.

Having said that I am not the biggest fans of wrappers upon wrappers upon hooks upon wrappers. Thats not easy to follow imo

If you insist we need a jdugement from @AVHopp

Reply:
I have a slight tendency to one of the two approaches, but I can still be convinced in any direction 😉 I'll let the two of you discuss here first, if you need a third opinion, just ping me.

Reply:
Ive substantially simplified the base hook so that the `spaghetti` argument probably doesnt apply anymore, please reevaluate

Reply:
Reduction of `spaghettiness` much approved, doge happy, very sustain, such improve 🦮 

Ok, since I appreciate your efforts, I'd say I accept 😄 Note though that I've pushed one commit fixing some doc issues and adding a comment what the new code block does. (Promoting again the `gpf` push type, just in case)

Regarding the wrapping debate: well, that's the concept of function composition (when you use that term, it doesn't sound as bad 🙃). So if you say you are not a big fan, then you are now longer allowed to use `cattrs` :D Still curious what @AVHopp's tendency was, though?

And just one last comment: it might happen that exactly this code part will be redesigned (irrespective of your change) in case we ever make the switch to `cattrs` builtin tagged union approach.

Reply:
My opinion was that this should also take care of the abbreviations as I also do not like to hook hooks into hooks which are already hooked into by hooking hooks.